src/lib.rs

```diff
@@ -145,6 +156,39 @@ impl<'data> Iterator for DataFormatDescriptorIterator<'data> {
     }
 }
 
+struct KeyValueDataIterator<'data> {
+    data: &'data [u8],
+}
+
+impl<'data> Iterator for KeyValueDataIterator<'data> {
+    type Item = (&'data [u8], &'data [u8]);
```

Comment:
The spec guarantees that keys are UTF-8, so we could do `(&'data str, &'data [u8])` here, which seems friendlier. Thoughts?

Reply:
Both are good ideas!